docs/registry/README.md

```diff
@@ -0,0 +1,385 @@
+# Registry Proposal
+
+## Problem 
+
+As an `Event Consumer` I want to be able to discover the different event types that I can consume 
+from the different Brokers, without resorting to any OOB mechanism. 
+This is also known as the `discoverability` use case, and is the main focus of this proposal. 
+
+## Objective
+
+Design an **initial** version of the **Registry** for the **MVP** that can support discoverability of 
+the different event types that can be consumed from the eventing mesh. For details on the different user stories 
+that this proposal touches, please refer to the 
+[User stories and personas for Knative eventing](https://docs.google.com/document/d/15uhyqQvaomxRX2u8s0i6CNhA86BQTNztkdsLUnPmvv4/edit?usp=sharing) document.
+
+#### Out of scope
+
+- Registry to Registry communication. This doesn't seem needed for an MVP.
+- Security-related matters. Those are handled offline by the `Cluster Configurator`, e.g., the `Cluster Configurator` 
+takes care of setting up Secrets for connecting to a GitHub repo, and so on.
+- Registry synchronization with `Event Producers`. We assume that if new GitHub events are created by 
+GitHub after our cluster has been configured (i.e., our GitHub CRD Source installed) and the appropriate webhooks 
+have been created, we will need to create new webhooks (and update the GitHub CRD Source) if we want to listen for 
+those new events. Until doing so, those new events shouldn't be listed in the Registry. 
+Such task will again be in charge of the `Cluster Configurator`.
+
+## Requirements
+
+Our design revolves around the following core requirements:
+
+1. We should have a Registry per namespace to enforce isolation.
+2. The Registry should contain the event types that can be consumed from
+the eventing mesh in order to support the `discoverability` use case.
+If an event type is not ready for consumption, we should explicitly indicate so (e.g., if the Broker 
+is not ready).
+3. The event types stored in the Registry should contain all the required information 
+for a consumer to create a Trigger without resorting to some other OOB mechanism.
+
+
+## Proposed Ideas
+
+### EventType CRD
+
+We propose introducing a namespaced-EventType CRD. Here is an example of how a CR would look like:
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: EventType
+metadata:
+  name: repopush
+  namespace: default
+spec:
+  type: repo:push
+  source: my-user/my-repo
+  schema: /my-schema
+  broker: default
+```
+
+- The `name` of the EventType is advisory, non-authoritative. Given that Cloud Event types can 
```

Comment:
Minor thing but s/Cloud Event/CloudEvents/   (no space and add an 's')

Reply:
will do, thanks!